src/Container/Container.php

```diff
@@ -0,0 +1,465 @@
+<?php
+
+/**
+ * This file is part of the Zest Framework.
+ *
+ * @author   Muhammad Umer Farooq <lablnet01@gmail.com>
+ * @author-profile https://www.facebook.com/Muhammadumerfarooq01/
+ *
+ * For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ *
+ * @since 3.0.0
+ *
+ * @license MIT
+ */
+
+namespace Zest\Container;
+
+use Zest\Contracts\Container\Container as ContainerContract;
+
+class Container implements ContainerContract
+{
+	/**
+	 * Registered type hints.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array
+	 */
+	protected $hints = [];
+
+	/**
+	 * Aliases.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array
+	 */
+	protected $aliases = [];
+
+	/**
+	 * Singleton instances.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array
+	 */
+	protected $instances = [];
+
+	/**
+	 * Contextual dependencies.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array
+	 */
+	protected $contextualDependencies = [];
+
+	/**
+	 * Instance replacers.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @var array
+	 */
+	protected $replacers = [];
+
+	/**
+	 * Parse the hint parameter.
+	 *
+	 * @param  string|array $hint Type hint or array contaning both type hint and alias
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return string
+	 */
+	protected function parseHint($hint): string
```

Comment:
It looks like the `$hint` will be the `array` or `string`.

Please check the type. I think it should throw `InvalidArgumentException` if `$hint` type is invalid :).

Reply:
Yes, consider the following code
```php
	protected function parseHint($hint): string
	{
		if(Arrays::isReallyArray($hint))
		{
			list($hint, $alias) = $hint;

			$this->aliases[$alias] = $hint;
		} elseif (!preg_match("/a-zA-Z/i", $hint)) {
			throw new \InvalidArgumentException ("The {$hint} parameter should be array or string ".gettype($hind)." given", 500);
			
		}

		return $hint;
	}
```

Reply:
@Lablnet, thank you for your reply.

This code snippets have following problems:

- coding style is about `if` condition should be consistency.
- I think we should use the `is_string` to validate the `$hint` is string before passing to the `preg_match`.

Reply:
Okay thanks for your response

```php
	protected function parseHint($hint): string
	{
		if (is_string($hint) && preg_match("/a-zA-Z/i", $hint)) {
			return $hint;
		} elseif(Arrays::isReallyArray($hint)) {
			list($hint, $alias) = $hint;

			$this->aliases[$alias] = $hint;
		} else {
			throw new \InvalidArgumentException ("The {$hint} parameter should be array or string ".gettype($hind)." given", 500);
		}

		return $hint;
	}
```

Reply:
@peter279k  Thanks for your response, finally it become
```php
	protected function parseHint($hint): string
	{
		if (is_string($hint) && preg_match("/a-zA-Z/i", $hint)) {
			return $hint;
		} elseif(Arrays::isReallyArray($hint)) {
			list($hint, $alias) = $hint;
			$this->aliases[$alias] = $hint;

			return $hint;
		}

		throw new \InvalidArgumentException ("The {$hint} parameter should be array or string ".gettype($hind)." given", 500);
	}
```

Reply:
Fix typo: `.gettype($hind)`. It's `$hint` :).